src/v/cloud_roles/tests/fetch_credentials_tests.cc

```diff
@@ -522,3 +522,66 @@ FIXTURE_TEST(test_intermittent_error, fixture) {
     auto aws_creds = std::get<cloud_roles::aws_credentials>(c.value());
     BOOST_REQUIRE_EQUAL(aws_creds.access_key_id, "my-key");
 }
+
+FIXTURE_TEST(test_imdsv2_fallback, fixture) {
+    tee_wrapper wrapper(cloud_roles::clrl_log);
+
+    fail_request_if(
+      [](const auto& req) {
+          return req._url == imdsv2_token_url && req._method == "PUT";
+      },
+      http_test_utils::response{
+        .body = "not found",
+        .status = ss::http::reply::status_type::not_found});
+
+    when()
+      .request(cloud_role_tests::aws_role_query_url)
+      .then_reply_with(cloud_role_tests::aws_role);
+
+    when()
+      .request(cloud_role_tests::aws_creds_url)
+      .then_reply_with(cloud_role_tests::aws_creds);
+
+    listen();
+
+    ss::abort_source as;
+    std::optional<cloud_roles::credentials> c;
+
+    one_shot_fetch s(c);
+
+    auto refresh = cloud_roles::make_refresh_credentials(
+      model::cloud_credentials_source::aws_instance_metadata,
+      as,
+      s,
+      cloud_roles::aws_region_name{""},
+      address());
+
+    refresh.start();
+    auto deferred = ss::defer([&refresh] { refresh.stop().get(); });
+
+    tests::cooperative_spin_wait_with_timeout(10s, [&c] {
+        return c.has_value();
+    }).get();
```

Comment:
nit: could use `RPTEST_REQUIRE_EVENTUALLY` instead which catches the time out error and fails the test for you

Reply:
fixed